exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java

```diff
@@ -364,4 +401,35 @@ private static SchemaPlus rootSchema(SchemaPlus schema) {
     }
   }
 
+  private static class DrillRexBuilder extends RexBuilder {
+    private DrillRexBuilder(RelDataTypeFactory typeFactory) {
+      super(typeFactory);
+    }
+
+    @Override
+    public RexNode ensureType(
+        RelDataType type,
+        RexNode node,
+        boolean matchNullability) {
+      RelDataType targetType = type;
+      if (matchNullability) {
+        targetType = matchNullability(type, node);
+      }
+      if (targetType.getSqlTypeName() == SqlTypeName.ANY) {
+        return node;
+      }
+      if (!node.getType().equals(targetType)) {
+        if(!targetType.isStruct()) {
+          final RelDataType anyType = TypeInferenceUtils.createCalciteTypeWithNullability(
```

Comment:
why do you extend RexBuilder and override ensureType() here? Most of the code is identical in RexBuilder.ensureType, except for line 422 - 427. Are they Drill-specific logic? If the answer is no, then we should put this logic in Calcite, not in Drill.